src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1118,6 +1155,312 @@ Containers::Optional<ImageData2D> AssimpImporter::doImage2D(const UnsignedInt id
+    if(configuration().value<bool>("mergeAnimationClips")) return -1;
+
+    if(!_f->animationIndicesForName) {
+        _f->animationIndicesForName.emplace();
+        _f->animationIndicesForName->reserve(_f->scene->mNumAnimations);
+        for(std::size_t i = 0; i != _f->scene->mNumAnimations; ++i)
+            _f->animationIndicesForName->emplace(std::string(_f->scene->mAnimations[i]->mName.C_Str()), i);
+    }
+
+    const auto found = _f->animationIndicesForName->find(name);
+    return found == _f->animationIndicesForName->end() ? -1 : found->second;
+}
+
+std::string AssimpImporter::doAnimationName(UnsignedInt id) {
+    /* If the animations are merged, don't report any names */
+    if(configuration().value<bool>("mergeAnimationClips")) return {};
+    return _f->scene->mAnimations[id]->mName.C_Str();
+}
+
+namespace {
+
+Animation::Extrapolation extrapolationFor(aiAnimBehaviour behaviour) {
+    switch(behaviour) {
+        case aiAnimBehaviour_DEFAULT:
+            /** @todo emulate this correctly by inserting keyframes
+                with the default node transformation */
+            return Animation::Extrapolation::Constant;
+        case aiAnimBehaviour_CONSTANT:
+            return Animation::Extrapolation::Constant;
+        case aiAnimBehaviour_LINEAR:
+            return Animation::Extrapolation::Extrapolated;
+        case aiAnimBehaviour_REPEAT:
+            return Animation::Extrapolation::Constant;
+        default: CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+    }
+}
+
+}
+
+Containers::Optional<AnimationData> AssimpImporter::doAnimation(UnsignedInt id) {
+    /* Import either a single animation or all of them together. At the moment,
+       Blender doesn't really support cinematic animations (affecting multiple
+       objects): https://blender.stackexchange.com/q/5689. And since
+       https://github.com/KhronosGroup/glTF-Blender-Exporter/pull/166, these
+       are exported as a set of object-specific clips, which may not be wanted,
+       so we give the users an option to merge them all together. */
+    const std::size_t animationBegin =
+        configuration().value<bool>("mergeAnimationClips") ? 0 : id;
+    const std::size_t animationEnd =
+        configuration().value<bool>("mergeAnimationClips") ? _f->scene->mNumAnimations : id + 1;
+
+    /* Calculate total channel count */
+    std::size_t channelCount = 0;
+    for(std::size_t a = animationBegin; a != animationEnd; ++a) {
+        channelCount += _f->scene->mAnimations[a]->mNumChannels;
+    }
+
+    /* Presence of translation/rotation/scaling keys for each channel.
+       We might skip certain tracks (see comments in the next loop)
+       but we need to know the correct track count and data array size
+       up front, so determine this and remember it for the actual
+       loop that extracts the tracks. */
+    Containers::Array<std::tuple<bool, bool, bool>> channelTrackPresence{channelCount};
```

Comment:
Is there a better way to do this? I tried `EnumSet` but `AnimationTrackTargetType` entries aren't binary exclusive.

Reply:
`AnimationTrackTargetType` could work with `BigEnumSet`, which only requires the enum to have some sane upper bound. 

Another alternative would be `BoolVector3` but its constructor is quite shitty and doesn't allow you to construct from bools directly (I suppose that's because I had a brain failure when trying to make that constexpr.)

Reply:
I settled for `BigEnumSet`, works great